plugin/pkg/admission/podnodeselector/admission.go

```diff
@@ -238,21 +235,26 @@ func (p *podNodeSelector) getNodeSelectorMap(namespace *api.Namespace) (labels.S
 	labelsMap := labels.Set{}
 	var err error
 	found := false
+	stableAnnotation := algorithm.AnnotationNamespaceNodeSelector
+	deprecatedAnnotation := algorithm.DeprecatedAnnotationNamespaceNodeSelector
 	if len(namespace.ObjectMeta.Annotations) > 0 {
-		for _, annotation := range NamespaceNodeSelectors {
-			if ns, ok := namespace.ObjectMeta.Annotations[annotation]; ok {
-				labelsMap, err = labels.ConvertSelectorToLabelsMap(ns)
-				if err != nil {
-					return labels.Set{}, err
-				}
-
-				if labels.Conflicts(selector, labelsMap) {
-					nsName := namespace.ObjectMeta.Name
-					return labels.Set{}, fmt.Errorf("%s annotations' node label selectors conflict", nsName)
-				}
-				selector = labels.Merge(selector, labelsMap)
-				found = true
+		ns, ok := namespace.ObjectMeta.Annotations[stableAnnotation]
+		if !ok {
+			ns, ok = namespace.ObjectMeta.Annotations[deprecatedAnnotation]
+		}
+
+		if ok {
```

Comment:
why not use `if else` :)

Reply:
I reuse `ok`, it's a different meaning.

Reply:
The block in `if ok {` needs to execute if either of the parses succeed (first trying stable, then alpha). Changing it to `else` would be logically different and would cause the real logic to only execute if the stable annotation parses correctly.

Reply:
I see that, thanks for clarify.